src/Xamarin.Android.Build.Tasks/Xamarin.Android.Application.targets

```diff
@@ -35,4 +35,29 @@ Copyright (C) 2019 Microsoft Corporation. All rights reserved.
     </GetAndroidPackageName>
     <Exec Command="&quot;$(AdbToolPath)adb&quot; $(AdbTarget) shell am force-stop &quot;$(_AndroidPackage)&quot;" />
   </Target>
+  <PropertyGroup>
+    <AndroidAotCustomProfile Condition=" '$(AndroidAotCustomProfile)' == '' ">custom.aprof</AndroidAotCustomProfile>
+    <AndroidAotProfilerPort Condition=" '$(AndroidAotProfilerPort)' == '' ">9999</AndroidAotProfilerPort>
+    <_BeginAotProfilingDependsOnTargets>
+      _SetupAotProfiling;
+      Install;
+      _SetAotProfilingPropsOnDevice;
+      StartAndroidActivity;
+    </_BeginAotProfilingDependsOnTargets>
+  </PropertyGroup>
+  <Target Name="_SetupAotProfiling">
+    <PropertyGroup>
+      <AndroidEmbedProfilers>aot</AndroidEmbedProfilers>
+    </PropertyGroup>
+  </Target>
+  <Target Name="_SetAotProfilingPropsOnDevice">
+    <Exec Command="&quot;$(AdbToolPath)adb&quot; $(AdbTarget) shell setprop debug.mono.profile aot:port=$(AndroidAotProfilerPort)" />
+  </Target>
+  <Target Name="BuildAndStartAotProfiling"
+      DependsOnTargets="$(_BeginAotProfilingDependsOnTargets)">
+  </Target>
+  <Target Name="FinishAotProfiling"
+      DependsOnTargets="_ResolveSdks">
+    <Exec Command="&quot;$(MonoAndroidBinDirectory)aprofutil&quot; -s -v -f -p $(AndroidAotProfilerPort) -o &quot;$(AndroidAotCustomProfile)&quot;" />
+  </Target>
```

Comment:
Does the new target need a `$(AProfUtilExtraArgs)` property? Are there some flags people might want to add in the future like `--verbose`, etc.?

Reply:
Good idea, I named it `$(AProfUtilExtraOptions)` to be similar to `$(AndroidExtraAotOptions)`.